### PR TITLE
test(evmstaking/keeper): add test cases for withdrawal queue

### PR DESCRIPTION
**Pull request by zsystm**
Increased test coverage of [client/x/evmstaking/keeper/withdrawal_queue.go](https://github.com/piplabs/story/blob/main/client/x/evmstaking/keeper/withdrawal_queue.go) from 0% to 78.8%
- **What's Covered**
  - All application logic written on the Story side, excluding the SDK.
- **What’s Not Covered (Edge Cases Related to the SDK)**
  - Failed to get paramstore: this could occur if the param store is broken or not initialized properly.
  - Failures related to collections: this includes potential failures during KV store iteration.

**changes:**
- `s.addrs`: create accounts should be called after setting bech32 prefix correctly, unless it has cosmos prefix.
- `accountKeeper.EXPECT().GetModuleAddress`
  - it is more natural for returning fixed module addresses instead of randomly created account's addresses.
- `codec`: consensusAddrCodec must be prefixed with `storycons`, not `storyvaloper`

issue: #64

### CODE REVIEW

File: client/x/evmstaking/keeper/keeper_test.go [hunk 1]
```diff
@@ -78,16 +77,18 @@ func (s *TestSuite) SetupTest() {
 	cfg.SetBech32PrefixForAccount("story", "storypub")
 	cfg.SetBech32PrefixForValidator("storyvaloper", "storyvaloperpub")
 	cfg.SetBech32PrefixForConsensusNode("storyvalcons", "storyvalconspub")
+	// it should be called after setting the bech32 prefix correctly
+	s.addrs = simtestutil.CreateIncrementalAccounts(4)
```
Comment (limengformal) [inline client/x/evmstaking/keeper/keeper_test.go:81]: Is this being used after your modification?

File: client/x/evmstaking/keeper/withdrawal_queue_test.go [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+				s.Equal(tc.expected[i].Amount, w.Amount)
+			}
+		})
+	}
+}
+
+func (s *TestSuite) TestGetAllWithdrawals() {
+	s.initQueue()
+
+	// Add a withdrawal to the queue
```
Comment (limengformal) [inline client/x/evmstaking/keeper/withdrawal_queue_test.go:187]: Add a test to withdraw for an empty queue

File: client/x/evmstaking/keeper/withdrawal_queue_test.go [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+			maxDequeue:  10,
+			expectedLen: len(withdrawals),
+			expected:    withdrawals,
+		},
+		{
+			name:        "Dequeue with empty queue",
+			maxDequeue:  3,
+			expectedLen: 0,
+			expected:    []types.Withdrawal{},
+		},
```
Comment (limengformal) [inline client/x/evmstaking/keeper/withdrawal_queue_test.go:78]: Add dequeue more than MaxWithdrawalPerBlock

File: client/x/evmstaking/keeper/withdrawal_queue_test.go [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+			maxDequeue:  10,
+			expectedLen: len(withdrawals),
+			expected:    withdrawals,
+		},
+		{
+			name:        "Peek with empty queue",
+			maxDequeue:  3,
+			expectedLen: 0,
+			expected:    []types.Withdrawal{},
+		},
```
Comment (limengformal) [inline client/x/evmstaking/keeper/withdrawal_queue_test.go:148]: Add peek a queue with items more than MaxWithdrawalPerBlock

File: client/x/evmstaking/keeper/keeper_test.go [hunk 5]
```diff
@@ -78,16 +77,18 @@ func (s *TestSuite) SetupTest() {
...
 	// gomock initializations
 	ctrl := gomock.NewController(s.T())
 
 	// mock keepers
 	accountKeeper := estestutil.NewMockAccountKeeper(ctrl)
-	accountKeeper.EXPECT().GetModuleAddress(types.ModuleName).Return(s.addrs[0]).AnyTimes()
-	accountKeeper.EXPECT().GetModuleAddress(stypes.ModuleName).Return(s.addrs[1]).AnyTimes()
-	accountKeeper.EXPECT().GetModuleAddress(stypes.BondedPoolName).Return(s.addrs[2]).AnyTimes()
-	accountKeeper.EXPECT().GetModuleAddress(stypes.NotBondedPoolName).Return(s.addrs[3]).AnyTimes()
+	accountKeeper.EXPECT().GetModuleAddress(types.ModuleName).Return(authtypes.NewModuleAddress(types.ModuleName)).AnyTimes()
```
Comment (limengformal) [inline client/x/evmstaking/keeper/keeper_test.go:88]: what are we testing here?

File: client/x/evmstaking/keeper/keeper_test.go (hunk 1 above)
Comment (zsystm) [reply client/x/evmstaking/keeper/keeper_test.go:81]: @limengformal 
`s.addrs` was not originally used anywhere. Since it was added by the story team, I tried to leave it untouched as much as possible, considering there might have been an intended purpose for this code. However, since it’s not being used, should we just remove it?

File: client/x/evmstaking/keeper/withdrawal_queue_test.go (hunk 2 above)
Comment (zsystm) [reply client/x/evmstaking/keeper/withdrawal_queue_test.go]: @limengformal 
Applied at https://github.com/piplabs/story/pull/61/commits/fdb9ab6306c7bc1382a0b12012d3df1ee4b78737

File: client/x/evmstaking/keeper/withdrawal_queue_test.go (hunk 3 above)
Comment (zsystm) [reply client/x/evmstaking/keeper/withdrawal_queue_test.go:84]: @limengformal 
Just to clarify, the `DequeueEligibleWithdrawals` is already designed to respect the `MaxWithdrawalPerBlock` limit. The test cases ensure that even if the queue has more withdrawals than the max limit, only up to `MaxWithdrawalPerBlock` are dequeued.

Could you please provide more details or an example of what you mean by ‘dequeue more than MaxWithdrawalPerBlock’? I want to ensure I’m addressing your concern accurately.

File: client/x/evmstaking/keeper/withdrawal_queue_test.go (hunk 4 above)
Comment (zsystm) [reply client/x/evmstaking/keeper/withdrawal_queue_test.go:155]: @limengformal 
ditto with https://github.com/piplabs/story/pull/61#discussion_r1733641549

File: client/x/evmstaking/keeper/keeper_test.go (hunk 5 above)
Comment (zsystm) [reply client/x/evmstaking/keeper/keeper_test.go:88]: @limengformal 
To clarify your question, do you mean what is the purpose of these lines of code?

These mocked `GetModuleAddress` calls are part of the necessary setup for the staking keeper initialization during the test. While the return values themselves are not directly asserted in the test, they simulate the real behavior of the account keeper methods, ensuring that the keeper can properly initialize and interact with the relevant modules. 

I just updated these mocks to reflect more accurately the actual module addresses that would be returned during execution, even though the specific addresses aren’t the focus of the test.

File: client/x/evmstaking/keeper/withdrawal_queue_test.go (hunk 3 above)
Comment (limengformal) [reply client/x/evmstaking/keeper/withdrawal_queue_test.go:84]: Sounds good. I missed that. Looks good now

File: client/x/evmstaking/keeper/keeper_test.go (hunk 1 above)
Comment (limengformal) [reply client/x/evmstaking/keeper/keeper_test.go:81]: If not used, let's remove it. I think it was originally used to test the removed mocked GetModuleAddress calls.

Comment (zsystm) [reply client/x/evmstaking/keeper/keeper_test.go]: Removed at https://github.com/piplabs/story/pull/61/commits/8112fb03a6fbb6fbd9fe17f06a4523b53f002137

**Review by limengformal — APPROVED** [review]
